osu.Game/Screens/SelectV2/SoloSongSelect.cs

```diff
@@ -26,6 +29,18 @@ public void Edit(BeatmapInfo beatmap)
             this.Push(new EditorLoader());
         }
 
+        /// <summary>
+        /// Opens results screen with the given score.
+        /// This assumes active beatmap and ruleset selection matches the score.
+        /// </summary>
+        public void PresentScore(ScoreInfo score)
+        {
+            Debug.Assert(Beatmap.Value.BeatmapInfo.Equals(score.BeatmapInfo));
+            Debug.Assert(Ruleset.Value.Equals(score.Ruleset));
+
+            this.Push(new SoloResultsScreen(score));
+        }
```

Comment:
This skips a step in old song select where beatmap and ruleset selection is forced to the score, it doesn't seem to be required here as the score cannot possibly be visible without the beatmap and ruleset matching.

Reply:
I don't know about that, but we'll see.